pbf/views.py

```diff
@@ -237,13 +247,7 @@ def add_player(request, game_id):
     if '-' in spirit_name:
         spirit_name, aspect = spirit_name.split(' - ')
     spirit = get_object_or_404(Spirit, name=spirit_name)
-    
-    if aspect in spirit_starting_energy.keys():
-        starting_energy = spirit_starting_energy[aspect]
-    else:
-        starting_energy = spirit_starting_energy[spirit.name]
-
-    gp = GamePlayer(game=game, spirit=spirit, color=colors[0], aspect=aspect, starting_energy=starting_energy)
+    gp = GamePlayer(game=game, spirit=spirit, color=colors[0], aspect=aspect, energy=spirit_setup_energy.get(spirit_name + (("/" + aspect) if aspect else ""), 0), starting_energy=spirit_base_energy_per_turn[spirit.name])
```

Comment:
I just know if this gets committed, the next expansion will have an aspect that customizes the energy track of a spirit; I'm in favor of leaving the logic in place, although I like the change you made to make it a little more clear which aspects go with which spirits.

Maybe both `energy` and `starting_energy` could be defined as local variables above `gp` doing the same one line for aspects?

Reply:
this whole line is getting a little long, so I think I'm on board with moving those variables out.

Depending on which spirit gets a modified energy track, `starting_energy` might not be able to be written as a one-liner, because let's say River gets a new aspect (let's pretend it's called Flowing) that modifies its tracks. Flowing would get added to `spirit_base_energy_per_turn`, and we'd need to look up Flowing if the aspect is Flowing, but River for all other aspects and base. Based on this I posit that it's difficult to say what form the code for it should take. By YAGNI I think I should leave it out, but I could add a comment that helps explain.